src/main/java/org/folio/service/transactions/BatchTransactionService.java

```diff
@@ -81,4 +136,106 @@ private Future<Void> patchTransactions(List<TransactionPatch> patches, DBConn co
     return Future.failedFuture(new HttpException(500, "transactionPatches: not implemented"));
   }
 
+  private void populateMetadata(List<Transaction> transactions, Map<String, String> okapiHeaders) {
+    // NOTE: Okapi usually populates metadata when a POST method is used with an entity with metadata.
+    // But in the case of the batch API there is no top-level metadata, so it is not populated automatically.
+    for (Transaction tr : transactions) {
+      if (tr.getMetadata() == null) {
+        String userId = okapiHeaders.get("X-Okapi-User-Id");
```

Comment:
we can use XOkapiHeaders to get constant names

Reply:
OK